### PR TITLE
eduction insitution data preprocessing

### PR DESCRIPTION
**Pull request by yuiint**
Preprocessing

- elementary school
- junior high school
- senior high school
- university

### CODE REVIEW

**Comment by AbnerTeng** [comment]
@yuiint please view the comment, tks!

File: src/preprocessing_edu.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    # sum the number of students  and conserve only useful columns
+    data = data.groupby('學校名稱', as_index=False).agg({ '縣市名稱': 'first', '總計': 'sum', 'lat': 'first', 'lng': 'first'})
+    data = data.rename(columns={'總計': '學生總數'})
+    data['縣市名稱'] = data['縣市名稱'].str.replace(r'\d| ', '', regex=True)
+    data = add_twd97_coordinates_to_dataframe(data)
+
+    return data
+
+
+def preprocessing_SHS() -> pd.DataFrame:
```
Comment (AbnerTeng) [inline src/preprocessing_edu.py:31]: in PEP8 style, you **cannot** use capital name. Example: `preprocessing_shs`

File: src/preprocessing_edu.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    """
+    Preprocess senior high school data
+    """
+    SHS_data_path = os.path.join('data', 'external_data')
+    SHS_data_path = os.path.join(SHS_data_path, '高中基本資料.csv')
```
[the remark this answers is not in the document]
Comment (AbnerTeng) [reply src/preprocessing_edu.py:36]: `shs_data_path`

File: src/preprocessing_edu.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    data = data[['學校名稱', '縣市名稱', '教師總數', '學生總數', 'lat', 'lng', '橫坐標', '縱坐標']]
+    
+    return data
+
+
+def preprocessing_JHS() -> pd.DataFrame:
+    """
+    Preprocess junior high school data
+    """
+    JHS_data_path = os.path.join('data', 'external_data')
```
Comment (AbnerTeng) [inline src/preprocessing_edu.py:55]: `jhs_data_path`

File: src/preprocessing_edu.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    JHS_data_path = os.path.join(JHS_data_path, '國中基本資料.csv')
```
[the remark this answers is not in the document]
Comment (AbnerTeng) [reply src/preprocessing_edu.py:56]: `jhs_data_path`

File: src/preprocessing_edu.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    data = load_data(JHS_data_path)
```
Comment (AbnerTeng) [inline src/preprocessing_edu.py:58]: `jhs_data_path`

File: src/preprocessing_edu.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                     data['學生數8年級女'] + data['學生數8年級男'] + data['學生數8年級女']
+    
+    # Conserve only useful columns
+    # data = data[['金融機構名稱','地址', 'lat', 'lng']]
+    data = data[['學校名稱', '縣市名稱', '教師總數', '學生總數', 'lat', 'lng', '橫坐標', '縱坐標']]
+
+    return data
+
+
+def preprocessing_ES() -> pd.DataFrame:
```
Comment (AbnerTeng) [inline src/preprocessing_edu.py:73]: `preprocessing_es`

File: src/preprocessing_edu.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+    """
+    Preprocess junior high school data
+    """
+    ES_data_path = os.path.join('data', 'external_data')
```
Comment (AbnerTeng) [inline src/preprocessing_edu.py:77]: `es_data_path`

File: src/preprocessing_edu.py [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+    ES_data_path = os.path.join(ES_data_path, '國小基本資料.csv')
```
[the remark this answers is not in the document]
Comment (AbnerTeng) [reply src/preprocessing_edu.py:78]: `es_data_path`

File: src/preprocessing_edu.py [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+    data = load_data(ES_data_path)
```
Comment (AbnerTeng) [inline src/preprocessing_edu.py:80]: `es_data_path`

**Comment by yuiint** [comment]
@AbnerTeng Done!!!

**Review by AbnerTeng — APPROVED** [review]
Good!!!

**Pull request merged**